lib/requester.js

```diff
@@ -71,11 +70,17 @@ function getAccessToken(operation, callback){
 
   const req = https.request(options, (res) => {
     res.on('data', (d) => {
-      operation.accessToken = JSON.parse(d.toString()).access_token;
-      callback(operation);
+      const responseValue = JSON.parse(d.toString());
+      operation.accessToken = responseValue.access_token;
+      operation.expiration = new Date(responseValue[".expires"]);
+      if(operation.lockAccessToken){
+        operation.lockAccessToken = false;
+      }
+      authenticatedRequest(operation);
     });
     res.on('error', (e) => {
       operation.accessToken = new Error(e);
+      throw new Error(operation.accessToken);
```

Comment:
I think that's worth a comment to explain what's going on there - e.g.

    // Throw an error that the MarkLogic Cloud authentication code will recognize

I'm not clear though on how that works - line 304 is doing this:

    if (operation.accessToken instanceof Error)
        throw new Error(operation.accessToken);

If an error is now thrown, then under what condition would `operation.accessToken` be an instance of Error?

Reply:
On line 82, operation.accessToken is set as the Error. So for a thread that came here (since it did not get the getAccessToken lock or for whatever reason), will be notified not to proceed with the operation.